src/strings-in-block/class-html.php

```diff
@@ -211,12 +211,28 @@ private function getTranslation( $text, $lang, \WP_Block_Parser_Block $block, ar
 		if ( $translationFromPageBuilder === $text ) {
 			$string_id = $this->get_string_id( $block->blockName, $text );
 			if ( Obj::path( [ $string_id, $lang, 'status' ], $string_translations ) == ICL_TM_COMPLETE ) {
-				return $string_translations[ $string_id ][ $lang ]['value'];
+				return self::preserveNewLines( $text, $string_translations[ $string_id ][ $lang ]['value'] );
 			} else {
 				return null;
 			}
 		} else {
 			return $translationFromPageBuilder;
 		}
 	}
+
+	private static function preserveNewLines( $original, $translation ) {
+		$endsWith = function ( $find, $s ) {
+			return Str::sub( - Str::len( $find ), $s ) === $find;
```

Comment:
@brucepearson, I was looking at the `Str::sub` and `Str::len` and noticed that we are not using multi-bytes functions. This is something that should be adjusted in the FP library I think, or do you know the reason for not using it?

Reply:
@strategio I've added a ticket - https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-8099